src/services/findAllReferences.ts

```diff
@@ -784,8 +784,8 @@ namespace ts.FindAllReferences.Core {
             return;
         }
 
-        const fullStart = state.options.findInComments || container.jsDoc !== undefined || forEach(search.symbol.declarations, d => d.kind === ts.SyntaxKind.JSDocTypedefTag);
-        for (const position of getPossibleSymbolReferencePositions(sourceFile, search.text, container, fullStart)) {
+        // Need to search in the full start of the node in case there is a reference inside JSDoc.
+        for (const position of getPossibleSymbolReferencePositions(sourceFile, search.text, container, /*fullStart*/ true)) {

```

Comment:
That was a bad test in the first place, because if you `container` is the SourceFile, `container.jsDoc` will always be undefined, but calling `.getStart()` on the SourceFile could still skip past jsdoc comments.
We should probably never even bother with calling `.getStart()` in `getPossibleSymbolReferencePositions` anyway, because scanning to the start of a node is probably more expensive than just doing text search on the full range.